src/r_data/models.cpp

```diff
@@ -457,6 +458,72 @@ static void ParseModelDefLump(int Lump)
 						Printf("%s: model not found in %s\n", sc.String, path.GetChars());
 					}
 				}
+				else if (sc.Compare("inherits"))
+				{
+					unsigned int i, j;
+					sc.MustGetString();	
+					auto type2 = PClass::FindClass(sc.String);
+					if (!type2 || type2->Defaults == nullptr)
+					{
+						sc.ScriptError("MODELDEF: Unknown actor type '%s'\n", sc.String);
+					}
+					for (i = 0; i < preParseFrames; i++)
+					{		
+						if (SpriteModelFrames[i].type == type2)
+						{		
+							auto frame = SpriteModelFrames[i];
+							frame.type = type;
+							SpriteModelFrames.Push(frame);
```

Comment:
`Push()` returns the index of the entry newly-pushed onto the array. You can save that value and avoid repeatedly calculating `SpriteModelFrames.Size() - 1`. Or better, because you're making a copy of the frame anyway, just alter that and push it at the end instead of the beginning.